common/lib/dependabot/shared_helpers.rb

```diff
@@ -58,13 +58,14 @@ def self.in_a_temporary_directory(directory = "/")
     end
 
     class HelperSubprocessFailed < StandardError
-      attr_reader :error_class, :error_context
+      attr_reader :error_class, :error_context, :trace
 
-      def initialize(message:, error_context:, error_class: nil)
+      def initialize(message:, error_context:, error_class: nil, trace: nil)
         super(message)
         @error_class = error_class || ""
         @error_context = error_context
         @command = error_context[:command]
+        @trace = trace
```

Comment:
Could we also set the backtrace here to have it rendered in sentry?

```suggestion
        @trace = trace
        set_backtrace(trace) if trace
```

Reply:
We can, but it would obfuscate the backtrace of the calling context, I think it might be more useful to keep the backtrace of the calling code in tact as well.

Reply:
Good point 👍